lib/internal/http2/core.js

```diff
@@ -2523,7 +2524,10 @@ class ServerHttp2Stream extends Http2Stream {
       this[kState].flags |= STREAM_FLAGS_HAS_TRAILERS;
     }
 
-    validateNumber(fd, 'fd');
+    if (fd instanceof fsPromisesInternal.FileHandle)
```

Comment:
isn't `instanceof` a bit slow? Maybe this world be better if the checks were the other way around?

Reply:
This was changed on request by @cjihrig … tbh, I wouldn’t expect it to make much of a difference in the big picture. And at some point in the recent past `instanceof` also became faster than a direct prototype comparison (according to 147b9d9792fe5772b554f4be77305805f361a42d), so it’s probably not too slow.